tealer/teal/instructions/instructions.py

```diff
@@ -83,6 +84,10 @@ def mode(self) -> ContractType:
         """Type of smart contracts this instruction is supported."""
         return self._mode
 
+    def cost(self, contract_version: int) -> int:
```

Comment:
`Instruction.cost(version)` feels a bit weird for an AI, especially considering that this just call `instruction_cost`. I see two alternatives:

- We remove it, and call `instruction_cost` directly in `_set_basic_block_costs`
- We refactor the code such as each basic block has access to the `Teal` object, and the `Teal` object has the `version` info. From that we can have `Instruction.cost(self)`, which get the version with `self.bb.teal.version`. This function should probably be memoize so that we avoid the  `_combine_dicts` cost on each call

What do you think?

Reply:
Agree, I'm thinking of going with the first alternative. Should I?

Reply:
yes, I will create an issue with notes for the second one